spec/linkedlist.js

```diff
@@ -0,0 +1,339 @@
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.head.data).to.equal("Mercedes")
+    })
+  })
+
+  context('insertFirst()', () => {
+    it('Inserts nodes (with the provided data) to the front of the linked list and checks the head of the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.tail.data).to.equal("Tesla")
+    })
+  })
+
+  context('contains(input)', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.contains("Lexus")).to.equal(true)
+    })
+  })
+
+  context('contains(input)', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.contains("BMW")).to.equal(true)
+    })
+  })
+
+  context('contains(input)', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.contains("Lexus")).to.equal(true)
+    })
+  })
+
+  context('contains(input)', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.contains("Ford")).to.equal(false)
+    })
+  })
+
+  context('find(input)', () => {
+    it('Returns the first node containing the provided data, or -1 if not found', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Ford")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.find("Ford").data).to.equal("Ford")
+    })
+  })
+
+  context('find( input )', () => {
+    it('Returns -1 if node is not found', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.find("Ford")).to.equal(-1)
```

Comment:
you have 6 different input function tests separated out and a lot of repeating code:
context('contains(input)', () => {
    it('Determines whether or not the list contains the provided data', () => {
      const myLinkedList = new LinkedList()
      myLinkedList.insert("BMW")
      myLinkedList.insert("Tesla")
      myLinkedList.insert("Lexus")
      myLinkedList.insertFirst("Mercedes")
      expect(myLinkedList.contains("Lexus")).to.equal(true)
      expect(myLinkedList.contains("BMW")).to.equal(true)
      expect(myLinkedList.contains("Lexus")).to.equal(true)
      expect(myLinkedList.contains("Ford")).to.equal(false)
      expect(myLinkedList.find("Ford").data).to.equal("Ford")
      expect(myLinkedList.find("Ford")).to.equal(-1)
    })
  })
looks like the first and third test here is a duplicate but you can see that all these expects will still work and you will save a lot of space in your file by making the more concise and easier to read.